app/src/main/java/fr/free/nrw/commons/nearby/fragments/NearbyParentFragment.java

```diff
@@ -1238,6 +1237,9 @@ private void setMapMarkerActions(Marker selectedMarker) {
     @Override
     public void recenterMap(fr.free.nrw.commons.location.LatLng curLatLng) {
         if (curLatLng == null) {
+            if (!locationManager.isProviderEnabled("network")) {
```

Comment:
The user of locationManager needs to know the correct strings to give this method for it to be able to work. A method `isNetworkProviderEnabled` would be better from an API perspective, or if this method accepted an Enum

Reply:
So, instead of `isProviderEnabled` in `LocationServiceManager`, I should add a method `isNetworkProviderEnabled`? Should I also add one for GPS?

Reply:
Yes or create an enum
```
enum class LocationProvider{
GPS("gps"), Network("network")
}
```
```
setProvider(LocationProvider locationProvider){
return locationManager.isProviderEnabled(locationProvider.getProvider());
```

Enums are generally better for api design as you don't have to add a new method per LocationProvider, just a new case of Enum.

Here it is probably overkill as there are only the 2 providers if memory serves. Also there are definitely constants for these and you shouldn't be using raw strings

Reply:
Enums seem neat. I will keep those in mind for future implementations but you are also right that it's overkill here for just two providers.